### PR TITLE
update taproot-updates.md

### PR DESCRIPTION
**Pull request by antonilol**
- Save a byte in some scripts
- Fix a script
- Set sequence to 1 in second stage HTLC
- Remove trailing whitespace

see the review comments at the specific lines

### CODE REVIEW

File: taproot-updates.md [hunk 1]
```diff
@@ -96,8 +96,8 @@ The commitment transaction will then become:
           "tapleaf": "
             # funds go back to the other channel participant after 1 block
             <remote_pubkey>
-            OP_CHECKSIGVERIFY
-            1 OP_CHECKSEQUENCEVERIFY
+            OP_CHECKSIG
+            OP_CHECKSEQUENCEVERIFY
```
Comment (antonilol) [inline taproot-updates.md:100]: saves a byte without losing functionality

testnet example: 78f96c750aeaa028c48a876ee0c07d2746de08ec24a51d9f7de08257e1dcd5fc

File: taproot-updates.md [hunk 2]
```diff
@@ -137,14 +137,13 @@ The commitment transaction will then become:
             # funds go back to us via a second-stage HTLC-timeout transaction (which contains an absolute delay)
             # NB: we also need the remote signature, which prevents us from unilaterally changing the HTLC-timeout transaction
             <remote_htlcpubkey> OP_CHECKSIGVERIFY <local_htlcpubkey> OP_CHECKSIG
-            1 OP_CHECKSEQUENCEVERIFY
```
Comment (antonilol) [inline taproot-updates.md:140]: the remote signature already commits to the sequence of the second stage tx, so explicit `OP_CHECKSEQUENCEVERIFY` is not needed here, saving 2 bytes

File: taproot-updates.md [hunk 3]
```diff
@@ -161,15 +160,15 @@ The commitment transaction will then become:
...
           "tapleaf_2": "
             # funds go to the remote node after an absolute delay (timeout)
             <remote_htlcpubkey>
-            OP_CHECKSIGVERIFY
+            OP_CHECKSIG
+            OP_CHECKSEQUENCEVERIFY
             <cltv_expiry>
             OP_CHECKLOCKTIMEVERIFY
-            1 OP_CHECKSEQUENCEVERIFY
+            OP_DROP
```
Comment (antonilol) [inline taproot-updates.md:171]: this script would end with 2 top stack elements

this works, there are also other ways with scripts the same size but this one looks like the one earlier mentioned with a separate `<cltv_expiry> OP_CHECKLOCKTIMEVERIFY OP_DROP`

File: taproot-updates.md [hunk 4]
```diff
@@ -195,7 +194,7 @@ A taproot HTLC-success transaction looks like:
       "txid": "...",
       "vout": 42,
       "scriptSig": "<payment_preimage> <remotehtlcsig> <localhtlcsig>",
-      "sequence": 0
+      "sequence": 1
```
Comment (antonilol) [inline taproot-updates.md:197]: also without https://github.com/t-bast/lightning-docs/pull/19/files#r869579106, the sequence needs to be 1

File: taproot-updates.md [hunk 5]
```diff
@@ -284,13 +283,13 @@ structure (or something similar) to our commitment transaction:
     # funds go back to us via a second-stage PTLC-timeout transaction (which contains an absolute delay)
     # NB: we need the remote signature, which prevents us from unilaterally changing the PTLC-timeout transaction
     <remote_ptlcpubkey> OP_CHECKSIGVERIFY <local_ptlcpubkey> OP_CHECKSIG
-    1 OP_CHECKSEQUENCEVERIFY
+    OP_CHECKSEQUENCEVERIFY
```
Comment (antonilol) [inline taproot-updates.md:286]: same idea as https://github.com/t-bast/lightning-docs/pull/19/files#r869577855 but then with more `OP_CHECKSIG`s

File: taproot-updates.md [hunk 6]
```diff
@@ -284,13 +283,13 @@ structure (or something similar) to our commitment transaction:
...
   ",
   "tapleaf_2": "
     # funds go to the remote node via a second-stage Claim-PTLC-success transaction by completing an adaptor sig, revealing the payment secret
     # NB: we don't use musig2 here because it would force local and remote signatures to use the same sighash flags
     <local_ptlcpubkey> OP_CHECKSIGVERIFY <remote_ptlcpubkey> OP_CHECKSIG
```
Comment (antonilol) [inline taproot-updates.md:291]: why are there two tapleaves when the only difference between the two is the order? one can be removed to save 32 vbytes

File: taproot-updates.md [hunk 7]
```diff
@@ -284,13 +283,11 @@ structure (or something similar) to our commitment transaction:
     # funds go back to us via a second-stage PTLC-timeout transaction (which contains an absolute delay)
     # NB: we need the remote signature, which prevents us from unilaterally changing the PTLC-timeout transaction
     <remote_ptlcpubkey> OP_CHECKSIGVERIFY <local_ptlcpubkey> OP_CHECKSIG
-    1 OP_CHECKSEQUENCEVERIFY
```
Comment (antonilol) [inline taproot-updates.md:287]: same as https://github.com/t-bast/lightning-docs/pull/19/files#r869579106

**Comment by t-bast** [comment]
Thanks for the feedback!

I have one main comment though: I don't want the scripts here to be efficient, I want them to be as readable as possible. They can indeed be optimized (as you noticed), but this document should stay high-level(ish). Optimizing the scripts will be done with miniscript once we integrate Taproot in the bolts, that's where your proposal will make more sense.

Can you please remove the script optimizations and keep the other changes? Thanks :pray:

**Comment by antonilol** [comment]
done

**Review by t-bast — APPROVED** [review]
Thanks!

File: taproot-updates.md (hunk 6 above)
Comment (antonilol) [reply taproot-updates.md]: this is only for readability, right?

**Pull request merged**